boards/risc-v/qemu-rv/rv-virt/src/qemu_rv_appinit.c

```diff
@@ -209,7 +208,7 @@ void board_early_initialize(void)
 #ifdef CONFIG_BOARDCTL_POWEROFF
 int board_power_off(int status)
 {
-#ifdef CONFIG_BUILD_KERNEL
```

Comment:
Suggestion, let the ecall go through and do "reset" in nsbi code. One option might be to define reset simply as 

```
reset:
  tail __start
```

Maybe you can think of something more elegant still :)